homework/list.js

```diff
@@ -1,43 +1,53 @@
 'use strict';
 
-(function() {
+//the way of how to refactor list.js is the same as content.js 
+//repair some error in list.js
+// add promise into it
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
+  var ListManager = function(){
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector("#note-content-wrapper");
+  }
+
+  ListManager.prototype = {
+    start() {
+      this.fetchList().then((function(data) {
+      this.updateList(data);
+      this.drawList();
+      this.preloadFirstNote();
+    }).bind(this))
+    .catch((function () {
```

Comment:
**Serious issue**: you need to catch the error in order to **log and re-throw** it. The reason is logging can prevent the following chains ignore it and make troubles when you're debugging, while re-throw it to make sure the next chain will be interrupted, rather than to perform the actions which may lead to unexpected behaviors.